### DISCUSSION TITLE
[BUG] Set-PnPTenantSite randomly fails with "Cannot get site $URL"

### DISCUSSION THREAD

**Discussion by iloving** [pnp/powershell#2302] [General]
I have a script that creates a new sharepoint site.  One of the steps involves modifying some of the settings for the site, which I do via the following command:

`Set-PnPTenantSite -Connection $Connection -Identity "$SiteFullURL" -DenyAddAndCustomizePages:$false -DisableAppViews:Disabled -DisableFlows:Disabled -Wait -ErrorVariable result -ErrorAction Continue`

It will, completely randomly, fail with:

Set-PnPTenantSite: /_$PATH_/MasterSiteCreateScript.ps1:104:9
Line |
 104 |          Set-PnPTenantSite -Connection $Connection -Identity "$SiteFul …
     |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Cannot get site https://_$MY-URL_.

To get around it, I have had to put it into a loop where I retry the command several times.  

### Expected behavior 
I expect it to execute successfully.  The site most definitely exists.

### Actual behavior 
It randomly fails for unknown reasons.

### Steps to reproduce behavior

Pseudocode is:
1. Connect to admin site
2. Check if the target site already exists.  If so, delete it
3. Wait 5 seconds
4. Create site with ` New-PnPTenantSite -Url $SiteFullURL -Owner $SiteOwner -Title $SiteTitle -Template $SiteTemplate -TimeZone $TimeZone -RemoveDeletedSite -Wait
`
5. Wait 5 seconds
6. Reconnect directly to new site `$Connection = Connect-PnPOnline -Url $SiteFullURL -ReturnConnection -Credentials $Credential`
7. Update permissions for a couple of the groups, disable AppViews and Flows via PnP Context
8. Enable scripting and repeat disabling App Views and Flows because there multiple ways of doing this and I'm not sure which is the "correct" since everything changes so much.  `Set-PnPTenantSite -Connection $Connection -Identity "$SiteFullURL" -DenyAddAndCustomizePages:$false -DisableAppViews:Disabled -DisableFlows:Disabled -Wait -ErrorVariable result -ErrorAction Continue`

Step 8 fails randomly with the above error, despite the fact that I had very obviously created the site and had performed multiple operations against it already.  A subsquent attempt likely succeeds, so I've had to put that command into a while loop to retry.

### What is the version of the Cmdlet module you are running?
Confirmed with 1.10.0 and 1.11.0

### Which operating system/environment are you running PnP PowerShell on?
- [ ] Windows
- [ ] Linux
- [X] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by CallumCrowley** [comment]
Please set tracing to "debug" using Set-PnPTraceLog and provide further information. This will allow the community to better understand the issue.

**Comment by gautamdsheth** [comment]
Same here as well, will have to move this to discussions since this an issue with the environment itself and not the cmdlet.

**Comment by hajjkh** [comment]
I have the same issue. Randomly I get the error "Cannot get site https://$MY-URL". Is there some fix for this?

**Comment by AndersRask** [comment]
It looks like site provisioning isn't completely done, even though you set wait.
A workaround could be to set -ValidateConnection on you connection, place it inside a try/catch that swallows the exception (write a warning and set a sleep for 30 seconds) inside a loop that retries x times
